url85940d3d8e_ccp999_Expr4j_tgz-pJ8-org_boris_expr_util_ExprEvaluatorJ8/src/org/boris/expr/function/excel/AVEDEV.java

```diff
@@ -0,0 +1,81 @@
+package org.boris.expr.function.excel;
+
+import org.boris.expr.Expr;
+import org.boris.expr.ExprArray;
+import org.boris.expr.ExprDouble;
+import org.boris.expr.ExprError;
+import org.boris.expr.ExprEvaluatable;
+import org.boris.expr.ExprException;
+import org.boris.expr.ExprInteger;
+import org.boris.expr.ExprMissing;
+import org.boris.expr.ExprNumber;
+import org.boris.expr.ExprString;
+import org.boris.expr.IEvaluationContext;
+import org.boris.expr.function.AbstractFunction;
+
+public class AVEDEV extends AbstractFunction
+{
+    public Expr evaluate(IEvaluationContext context, Expr[] args)
+            throws ExprException {
+        assertMinArgCount(args, 1);
+
+        double[] values = { 0, 0 };
+
+        for (Expr a : args)
+            AVERAGE.eval(context, a, values, true);
+
+        if (values[1] == 0) {
+            return ExprError.NUM;
+        }
+
+        double average = values[0] / values[1];
+
+        values[0] = values[1] = 0;
+
+        for (Expr a : args)
+            eval(context, a, average, values, true);
+
+        return new ExprDouble(values[0] / values[1]);
+    }
+
+    public static void eval(IEvaluationContext context, Expr a, double average,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 66-77 into a separate method.